From ebcd5a44ee45ab5445ec4ae7821338332d8cccdd Mon Sep 17 00:00:00 2001
From: Sushrut Shree Trivedi <quic_sushruts@quicinc.com>
Date: Wed, 23 Jul 2025 16:03:23 +0530
Subject: [PATCH] PENDING: arm64: dts: qcom: Enable PCIe0 & PCIe1 for QCS8275
 IQ8 EVK

Enable PCIe0 and PCIe1 for QCS8275 IQ8 EVK.

Signed-off-by: Sushrut Shree Trivedi <quic_sushruts@quicinc.com>
Upstream-Status: Pending
---
 .../boot/dts/qcom/qcs8275-iq-8275-evk.dts     | 80 +++++++++++++++++++
 1 file changed, 80 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts b/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts
index add5ee3a5778..feb30c2ebc99 100644
--- a/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts
+++ b/arch/arm64/boot/dts/qcom/qcs8275-iq-8275-evk.dts
@@ -212,6 +212,40 @@ vreg_l9c: ldo9 {
 	};
 };
 
+&pcie0 {
+	perst-gpios = <&tlmm 2 GPIO_ACTIVE_LOW>;
+	wake-gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie0_default_state>;
+
+	status = "okay";
+};
+
+&pcie1 {
+	perst-gpios = <&tlmm 23 GPIO_ACTIVE_LOW>;
+	wake-gpios = <&tlmm 21 GPIO_ACTIVE_HIGH>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie1_default_state>;
+
+	status = "okay";
+};
+
+&pcie0_phy {
+	vdda-phy-supply = <&vreg_l6a>;
+	vdda-pll-supply = <&vreg_l5a>;
+
+	status = "okay";
+};
+
+&pcie1_phy {
+	vdda-phy-supply = <&vreg_l6a>;
+	vdda-pll-supply = <&vreg_l5a>;
+
+	status = "okay";
+};
+
 &camss {
 	status = "okay";
 };
@@ -400,6 +434,52 @@ dp_hot_plug_det: dp-hot-plug-det-state {
 	/* TODO: Add the required nodes once mdss_dp0 is available.*/
 	};
 
+	pcie0_default_state: pcie0-default-state {
+		perst-pins {
+			pins = "gpio2";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+
+		clkreq-pins {
+			pins = "gpio1";
+			function = "pcie0_clkreq";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+
+		wake-pins {
+			pins = "gpio0";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
+
+	pcie1_default_state: pcie1-default-state {
+		perst-pins {
+			pins = "gpio23";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+
+		clkreq-pins {
+			pins = "gpio22";
+			function = "pcie1_clkreq";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+
+		wake-pins {
+			pins = "gpio21";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
+
 	qup_i2c0_default: qup-i2c0-state {
 		pins = "gpio17", "gpio18";
 		function = "qup0_se0";
-- 
QUALCOMM INDIA, on behalf of Qualcomm Innovation Center, Inc. is a member 
of Code Aurora Forum, hosted by The Linux Foundation

